daemon/daemon.go

```diff
@@ -482,14 +529,47 @@ func (d *Daemon) Stop() error {
 		d.snapListener.Close()
 	}
 
+	if restartSystem {
+		// give time to polling clients to notice restart
+		time.Sleep(300 * time.Millisecond)
+	}
+
 	d.tomb.Kill(d.snapdServe.finishShutdown())
 	if d.snapListener != nil {
 		d.tomb.Kill(d.snapServe.finishShutdown())
 	}
 
+	if !restartSystem {
+		// tell systemd that we are stopping
+		systemdSdNotify("STOPPING=1")
+
+	}
+
 	d.overlord.Stop()
 
-	return d.tomb.Wait()
+	err := d.tomb.Wait()
+	if err != nil {
+		return err
+	}
+
+	if restartSystem {
+		// ask for shutdown and wait for it to happen.
+		// if we exit snapd will be restared by systemd
+		rebootDelay := "+1"
```

Comment:
Any reason to wait for a whole minute here?

Reply:
I had "now" at some point here, but at the last sprint discussing about this work with Tony he pointed out that things could go a bit too fast for somebody doing "snap refresh" from the terminal that refreshes kernel or core to understand what is going on. I don't have a super strong opinion we could try with "now"  and see what is people feedback. cc @mvo5